imap-proto/src/lib.rs

```diff
@@ -3,17 +3,13 @@ extern crate nom;
 
 #[macro_use]
 mod macros;
-mod parser;
 
-pub mod body;
-pub mod body_structure;
-pub mod builders;
-pub mod types;
+// Public API
```

Comment:
This commit has too many things going on. Please separate them into separate commits so I can review them separately. If your commit message has a list like this, it's a good sign that it should have each bullet as a separate commit (moving files and modules around can be all in one commit, I suppose).

Reply:
(Also should mention that I like the direction to avoid wildcard imports!)